File: easy/strings/merge_alt_strs.cpp
// https://leetcode.com/problems/merge-strings-alternately/description/?envType=study-plan-v2&envId=leetcode-75


class Solution {
    public:
        string mergeAlternately(string word1, string word2) {
            int i = 0;
            int t = 0;
            string str = "";
            int smallest = std::min(word1.length(), word2.length());
            while (i < smallest) {
                if (t == 0) {
                    str += word1[i];
                }
                else if (t == 1) {
                    str += word2[i];
                }
                t++;
                if (t == 2) {
                    t = 0;
                    i++;
                }
            }
    
            if (word1.length() == word2.length()) return str;
            // append onto end of merged string
            string trail = i < word1.length() ? word1 : word2;
            for (int m = i; m < trail.length(); ++m) {
                str += trail[m];
            }
            return str;
            
        }
    };

File: easy/strings/merge_alt_strs.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i = 0
        j = 0
        l1 = len(word1)
        l2 = len(word2)
        result = ""
        w1done = False
        w2done = False

        while True:
            if i >= l1 and j >=l2:
                w1done = True
                w2done = True
                break
            elif i >= l1:
                w1done = True
                break
            elif j >= l2:
                w2done = True
                break
            result += word1[i]
            result += word2[j]
            i += 1
            j += 1

        if w1done and not w2done:
            result += word2[j:]
        elif not w1done and w2done:
            result += word1[i:] 
        
        return result

        
        